projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-cell.component.ts

```diff
@@ -356,6 +356,10 @@ export class IgxGridFilteringCellComponent implements AfterViewInit, OnInit, DoC
     }
 
     private isColumnRightVisible(columnIndex: number): boolean {
+        // If displayContainerWidth is 0 than there is no horizontal scrollbar, which means that all columns are fully visible.
+        if (this.filteringService.displayContainerWidth === 0) {
```

Comment:
Could you create a property in filtering service to return true if all columns are visible. If someone in the future changes the behavior of `displayContainerWidth` this code will fail.